internal/resource/mappers/selectiontracker/selection_tracker.go

```diff
@@ -17,45 +17,51 @@ import (
 
 type WorkloadSelectionTracker struct {
 	lock     sync.Mutex
-	prefixes *radix.Tree[[]controller.Request]
-	exact    *radix.Tree[[]controller.Request]
+	prefixes *radix.Tree[[]*pbresource.ID]
+	exact    *radix.Tree[[]*pbresource.ID]
```

Comment:
Maybe these need to be `map[string(partition)]map[string(namespace)]radix.Tree[X]` or maybe make a flatTenancy struct with those two fields present and use that as a key?

Reply:
Yeah 🤦‍♀️ I made it use url-like string keys for paths constructed from tenancy. I ended up including peer name too. I'm not sure if we will actually need it for peering, but if we do, it will prevent us from conflating resources in local and non-local peers, although we will likely need to change this whenever we implement peer tenancy. 

Let me know if this makes sense to you.

Reply:
Yep. This is likely less work than the map juggling would be.